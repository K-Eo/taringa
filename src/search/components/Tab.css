.Tab-slate {
  margin: 10rem 0;
  text-align: center;
}
