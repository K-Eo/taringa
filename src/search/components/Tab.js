import React, { PureComponent } from 'react'
import { connect } from 'react-redux'

import './Tab.css'

export class Tab extends PureComponent {
  static defaultProps = {
    items: [],
    status: 'success',
  }

  render() {
    const { status, className, totalCount, items, children } = this.props

    const classes = className || 'card-body'
    const isFetching = status === 'fetching'
    const hasSuccess = status === 'success'
    const hasFailure = status === 'failure'
    const isInitial = hasSuccess && totalCount === undefined
    const hasResults = hasSuccess && totalCount !== undefined && totalCount > 0
    const isEmpty = hasSuccess && totalCount !== undefined && totalCount === 0

    if (hasResults) {
      return <div className={classes}>{children(items)}</div>
    }

    return (
      <div className="card-body">
        {isFetching && (
          <p className="Tab-slate">
            <i className="fa fa-circle-notch fa-spin mr-1" />
            Buscando...
          </p>
        )}

        {isInitial && (
          <p className="Tab-slate">Puedes buscar posts, usuarios o canales</p>
        )}

        {isEmpty && <p className="Tab-slate">No hay resultados</p>}

        {hasFailure && (
          <p className="Tab-slate text-danger">
            ¡Ratas! Intenta recargar la página
          </p>
        )}
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  console.log(state.searching[ownProps.id])
  return state.searching[ownProps.id] || {}
}

export default connect(mapStateToProps)(Tab)
