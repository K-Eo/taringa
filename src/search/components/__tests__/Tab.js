import React from 'react'

import Tab from '../Tab'

describe('Tab', () => {
  it('exists', () => {
    expect(Tab).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Tab />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders loader when is fetching', () => {
    const props = {
      status: 'fetching',
    }

    const wrapper = shallow(<Tab {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders children when has success', () => {
    const props = {
      children: 'foo',
      status: 'success',
      totalCount: 2,
    }

    const wrapper = shallow(<Tab {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders empty slate when has success and no items', () => {
    const props = {
      status: 'success',
      totalCount: 0,
    }

    const wrapper = shallow(<Tab {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders message when has failure', () => {
    const props = {
      status: 'failure',
    }

    const wrapper = shallow(<Tab {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
