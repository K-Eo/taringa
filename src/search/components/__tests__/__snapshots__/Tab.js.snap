// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders children when has success 1`] = `
<div
  className="card-body"
>
  foo
</div>
`;

exports[`Tab renders default 1`] = `
<div
  className="card-body"
>
  <p
    className="Tab-slate"
  >
    Puedes buscar posts, usuarios o canales
  </p>
</div>
`;

exports[`Tab renders empty slate when has success and no items 1`] = `
<div
  className="card-body"
>
  <p
    className="Tab-slate"
  >
    No hay resultados
  </p>
</div>
`;

exports[`Tab renders loader when is fetching 1`] = `
<div
  className="card-body"
>
  <p
    className="Tab-slate"
  >
    <i
      className="fa fa-circle-notch fa-spin mr-1"
    />
    Buscando...
  </p>
</div>
`;

exports[`Tab renders message when has failure 1`] = `
<div
  className="card-body"
>
  <p
    className="Tab-slate text-danger"
  >
    ¡Ratas! Intenta recargar la página
  </p>
</div>
`;
